src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/custom.py

```diff
@@ -200,6 +200,56 @@ def _server_georestore(cmd, client, resource_group_name, server_name, sku_name,
     return sdk_no_wait(no_wait, client.create, resource_group_name, server_name, parameters)
 
 
+# Custom functions for server replica, will add PostgreSQL part after backend ready in future
+def _replica_create(cmd, client, resource_group_name, server_name, source_server, no_wait=False, **kwargs):
+    provider = 'Microsoft.DBForMySQL' if isinstance(client, MySqlServersOperations) else 'Microsoft.DBforPostgreSQL'
+    # set source server id
+    if not is_valid_resource_id(source_server):
+        if len(source_server.split('/')) == 1:
+            source_server = resource_id(subscription=get_subscription_id(cmd.cli_ctx),
+                                        resource_group=resource_group_name,
+                                        namespace=provider,
+                                        type='servers',
+                                        name=source_server)
+        else:
+            raise CLIError('The provided source-server {} is invalid.'.format(source_server))
+
+    source_server_id_parts = parse_resource_id(source_server)
+    try:
+        source_server_object = client.get(source_server_id_parts['resource_group'], source_server_id_parts['name'])
+    except Exception as e:
```

Comment:
Any exception that cause the get server failed should be catched, (server not exists, or network/ARM/backend issue), because we need information of this server to continue.

Reply:
I hear you. I mean the python type. So you do not have a broad except. 

The mysqlserver operations [client](https://github.com/Azure/azure-sdk-for-python/blob/f8c6958695fed199b84f02f05f7df4fb5ac65a48/azure-mgmt-rdbms/azure/mgmt/rdbms/mysql/operations/servers_operations.py#L351) and postgresql operations [client](https://github.com/Azure/azure-sdk-for-python/blob/f8c6958695fed199b84f02f05f7df4fb5ac65a48/azure-mgmt-rdbms/azure/mgmt/rdbms/postgresql/operations/servers_operations.py#L351) both raise clouderror, when their get method fails.

So it might be good to catch `msrestazure.azure_exceptions.CloudError` instead of a broad Exception. Let me know your thoughts on this.

Reply:
I can change it to CloudError, then how about other exception? How will CLI framework handle other exceptions?

Reply:
Thanks, what other exceptions? You mean exceptions in general?

[Here](https://github.com/Azure/azure-cli/blob/bc7a5ae3be43bef3368fe19821dd62c358744c61/src/azure-cli-core/azure/cli/core/util.py#L22-L50) is the method the CLI calls to handle exceptions.

Reply:
Changed to CloudError in new commit.